src/libAtomVM/interop.c

```diff
@@ -138,6 +138,63 @@ char *interop_list_to_string(term list, int *ok)
     return str;
 }
 
+char *interop_list_to_utf8_string(term list, int *ok)
+{
+    size_t byte_len = 0;
+
+    term t = list;
+    while (term_is_nonempty_list(t)) {
+        term head = term_get_list_head(t);
+        if (UNLIKELY(!term_is_integer(head))) {
+            *ok = 0;
+            return NULL;
+        }
+        avm_int_t codepoint = term_to_int(head);
+        if (UNLIKELY(codepoint < 0)) {
+            *ok = 0;
+            return NULL;
+        } else if (codepoint <= 127) {
+            byte_len++;
+        } else {
+            size_t codepoint_size;
+            bitstring_utf8_encode(codepoint, NULL, &codepoint_size);
```

Comment:
Why are we not checking the return value from `bitstring_utf8_encode` here?  We do below, but only after we allocate the buffer.  So if the input term list is not a valid UTF8 string, we will have done an allocation and then later if it is invalid, we free the buffer and let the calling function know it's invalid (which is might be, as it is only being called from the `list_to_atom` Nif, where the list is user-supplied)

Reply:
we do the check few lines later, but I can do it earlier, fair enough.

Reply:
Fixed, earlier is better indeed.